### PR TITLE
fix(connector): minor fixes and improvements

### PR DESCRIPTION
**Pull request by johnsonogwuru**
This PR includes, changes related to setting the Authorization transactions to `success` only after the user confirms a transaction on paypal, and other tiny improvements around error handling.

### CODE REVIEW

File: processor/src/services/paypal-payment.service.ts [hunk 1]
```diff
@@ -221,6 +218,16 @@ export class PaypalPaymentService extends AbstractPaymentService {
 
     this.validateInterfaceIdMismatch(ctPayment, opts.data.orderId);
 
+    await this.ctPaymentService.updatePayment({
```
Comment (dasanorct) [inline processor/src/services/paypal-payment.service.ts:221]: we should not do this at this point because this means the payment is authorized and with an authorized payment an order can be created.
the paypal capture, which is the order confirmation in paypal, could fail and that would mean that we have an order created but a failure payment in paypal.

we are working with the CoCo payment events so every time we modify a payment CoCo throws an event and we process it in the same way in checkout regardless the PSP behind. what does this mean? right after you add an authorization transaction in the state pending or success checkout is going to create the order.

in summary, we have 2 options:

- we do not create an authorize transaction, just the charge transaction once we have the payment confirmation from paypal (order capture)
- we add the authorization transaction after the paypal captureorder call, either calling 2 times to the payments-sdk or modifying the method in the payments-sdk to accept an array of transactions

**Review by dasanorct — APPROVED** [review]
